src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -453,10 +453,7 @@ RateLimitSettings getRateLimitSettings() {
   static Map<String, ConfigValue> validateAll(final String topic, final Map<String, String> props) {
     String prefix = format("%s%s.", TOPIC_OVERRIDE_PREFIX, topic);
     List<String> topicOverrides =
-        props.keySet().stream()
-            .filter(k -> k.startsWith(prefix))
-            .map(k -> k.substring(0, prefix.length()))
-            .collect(Collectors.toList());
+        props.keySet().stream().filter(k -> k.startsWith(prefix)).collect(Collectors.toList());
```

Comment:
Can you explain how this fits together?  I see that the call to `map` is removed, but I'm not clear why.

Reply:
That was a bug, as the map step removed the name of the actual override.
We now keep the full original property name and report on that and its validity.

Reply:
To clarify some more, previously we just gave the name of the variable used (so the override prefix was removed) rather than the config the user provided.